pkg/reconciler/namespace/controller.go

```diff
@@ -0,0 +1,259 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+	corelister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/kcp-dev/kcp/pkg/apis/cluster/v1alpha1"
+	clusterclient "github.com/kcp-dev/kcp/pkg/client/clientset/versioned"
+	"github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	clusterlisters "github.com/kcp-dev/kcp/pkg/client/listers/cluster/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/gvk"
+	"github.com/kcp-dev/kcp/pkg/informer"
+)
+
+const resyncPeriod = 10 * time.Hour
+
+// NewController returns a new Controller which schedules namespaced resources to a Cluster.
+func NewController(cfg *rest.Config) *Controller {
+	dynClient := dynamic.NewForConfigOrDie(cfg)
+	kubeClient := kubernetes.NewForConfigOrDie(cfg)
+	clusterClient := clusterclient.NewForConfigOrDie(cfg)
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	stopCh := make(chan struct{}) // TODO: hook this up to SIGTERM/SIGINT
```

Comment:
@ncdc will not be happy with this one - are we missing some normal controller flow, why do we have this in all of our constructors? (I've only ever written with controller-runtime)

Reply:
Yeah I think I'm going to wait for #203 and follow that pattern, which is much better. I think that will get merged before this one anyway.

Reply:
This is done now.